crates/core/src/sql/compiler.rs

```diff
@@ -423,13 +425,48 @@ mod tests {
         // Assert index scan.
         assert_index_scan(
             ops.swap_remove(0),
-            1.into(),
+            ColId(1),
             Bound::Included(2u64.into()),
             Bound::Included(2u64.into()),
         );
         Ok(())
     }
 
+    #[test]
+    fn compile_index_multi_eq_and_eq() -> ResultTest<()> {
+        let (db, _) = make_test_db()?;
+        let mut tx = db.begin_mut_tx(IsolationLevel::Serializable);
+
+        // Create table [test] with index on [b]
+        let schema = &[
+            ("a", AlgebraicType::U64),
+            ("b", AlgebraicType::U64),
+            ("c", AlgebraicType::U64),
+            ("d", AlgebraicType::U64),
+        ];
+        let indexes = &[(0.into(), "a"), (1.into(), "b")];
```

Comment:
How many columns do we support? Also can we take advantage of a multi-column index in a join? It's fine if the answer is no, we just probably want to add another test if the answer is yes.

Reply:
> How many columns do we support?

We support up to 4 columns now. 

> We take advantage of a multi-column index in a join?

Internally, the `SQL` parser chokes with `on (a,b) = (d,c)` (not supported) and `ambiguous field` if used in a `where clause`.